packages/components/src/mobile/image/index.native.js

```diff
@@ -259,14 +289,57 @@ const ImageComponent = ( {
 					</View>
 				) : (
 					<View style={ focalPoint && styles.focalPointContent }>
-						<Image
-							style={ imageStyles }
-							source={ { uri: url } }
-							{ ...( ! focalPoint && {
-								resizeMethod: 'scale',
-							} ) }
-							resizeMode={ imageResizeMode }
-						/>
+						{ Platform.isAndroid && (
+							<>
+								{ networkImageLoaded && networkURL && (
+									<Animated.Image
+										style={ imageStyles }
+										fadeDuration={ 0 }
+										source={ { uri: networkURL } }
+										{ ...( ! focalPoint && {
+											resizeMethod: 'scale',
+										} ) }
+										resizeMode={ imageResizeMode }
+									/>
+								) }
+								{ ! networkImageLoaded && ! networkURL && (
+									<Animated.Image
+										style={ imageStyles }
+										fadeDuration={ 0 }
+										source={ { uri: localURL } }
+										{ ...( ! focalPoint && {
+											resizeMethod: 'scale',
+										} ) }
+										resizeMode={ imageResizeMode }
+									/>
+								) }
```

Comment:
The flickering still occurs when only one RNImage instance is used and the `source` prop is changed:

```
<Animated.Image
	style={ imageStyles }
	fadeDuration={ 0 }
	source={ { uri: networkURL ? networkURL : localURL } }
	{ ...( ! focalPoint && {
		resizeMethod: 'scale',
	} ) }
	resizeMode={ imageResizeMode }
/>
```

My understanding is that the flashing will always occur on Android when the `source` prop is changed. While it's very noticeable when using RNImage, it also affects FastImage on Android as well (as described in https://github.com/DylanVann/react-native-fast-image/issues/747), so it appears to be related to how image data is cached and retrieved on Android. The apparent solution is to use separate image instances to avoid changing the `source` prop entirely.